clients/context-registry/src/test/integration/ContextRegistryClient.test.ts

```diff
@@ -4,68 +4,121 @@
   });
 
-  it("should get a list of Most Recently Used (MRU) projects (#integration)", async () => {
-    const queryOptions: ContextRegistryRequestQueryOptions = {
-      $select: "*",
-      $top: 20,
-      isMRU: true,
-    };
+  it("should get a paged list of iTwins using top (#integration)", async () => {
+    const numberOfITwins = 3;
 
-    const projects: Project[] = await contextRegistry.getProjects(requestContext, queryOptions);
-    chai.expect(projects.length).greaterThan(0);
-  });
+    // Verify there are enough iTwins to test the paging
+    const fullITwinList: ITwin[] = await iTwinAccessClient.getAll(requestContext);
+    chai.assert(fullITwinList.length >= numberOfITwins, "Unable to meaningfully run test since there are too few iTwins.");
 
-  it("should get a list of Favorite projects (#integration)", async () => {
-    const queryOptions: ContextRegistryRequestQueryOptions = {
-      $select: "*",
-      $top: 20,
-      isFavorite: true,
-    };
+    const partialITwinList: ITwin[] = await iTwinAccessClient.getAll(requestContext,
+      {
+        pagination: {
+          top: numberOfITwins,
+        },
+      });
 
-    const projects: Project[] = await contextRegistry.getProjects(requestContext, queryOptions);
-    chai.expect(projects.length).to.be.greaterThan(0);
+    // Get the same number of iTwins as the top param
+    chai.expect(partialITwinList).length(numberOfITwins);
   });
 
-  it("should get a project by name (#integration)", async () => {
-    const queryOptions: ContextRegistryRequestQueryOptions = {
-      $select: "*",
-      $filter: `Name+eq+'${TestConfig.projectName}'`,
-    };
-    const project: Project = await contextRegistry.getProject(requestContext, queryOptions);
-    chai.expect(project.name).equals(TestConfig.projectName);
-  });
+  it("should get a paged list of iTwins using skip (#integration)", async () => {
+    const numberSkipped = 3;
+
+    // Verify there are enough iTwins to test the paging
+    const fullITwinList: ITwin[] = await iTwinAccessClient.getAll(requestContext);
+    chai.assert(fullITwinList.length >= numberSkipped, "Unable to meaningfully run test since there are too few iTwins.");
+
+    const partialITwinList: ITwin[] = await iTwinAccessClient.getAll(requestContext,
+      {
+        pagination: {
+          skip: numberSkipped,
+        },
+      });
 
-  it("should get a list of invited projects (#integration)", async () => {
-    const invitedProjects: Project[] = await contextRegistry.getInvitedProjects(requestContext);
-    chai.expect(invitedProjects.length).greaterThan(0);
+    // Get all but the skipped ones
+    chai.expect(partialITwinList).length(fullITwinList.length - numberSkipped);
   });
 
-  it("should get a team", async () => {
-    const team: Team = await contextRegistry.getTeam(requestContext);
-    chai.expect(team).not.to.be.undefined;
+  it("should get a continuous paged list of iTwins (#integration)", async () => {

```

Comment:
The title of this test would suggest that we get this kind of result (and that's what I would expect using the method):
| Params | Results |
|---|---|
| top: 8 | [1, 2, 3, 4, 5, 6, 7, 8]|
| top: 4 | [1, 2, 3, 4]|
| top: 4, skip: 4 | [5, 6, 7, 8]|

But the function will likely return this if we have only one source of iTwins for the page requested (Project Or Assets)

| Params | Results |
|---|---|
| top: 8 | [1, 2, 3, 4, 5, 6, 7, 8]| 
| top: 4 | [1, 2, 3, 4]|
| top: 4, skip: 4 | [3, 4, 5, 6]|

And this test will not catch this issue.

Reply:
I think I see the issue of skip you are referring to and I am investigating a fix. Fortunately the test does catch this issue since it verifies that the number of unique elements matches the number of elements skipped.

Reply:
A fix has been added that properly skips the correct elements.

Reply:
Marking this as resolved. You are welcome to check the code again to verify and reopen in needed.